rclpy/rclpy/action/client.py

```diff
@@ -463,14 +463,15 @@ def unblock(future: Future[Any]) -> None:
         send_goal_future.add_done_callback(unblock)
 
         event.wait()
-        exeception = send_goal_future.exception()
-        if exeception is not None:
-            raise exeception
+        exception = send_goal_future.exception()
+        if exception is not None:
+            raise exception
 
         goal_handle = send_goal_future.result()
 
-        if goal_handle is None:
```

Comment:
@fujitatomoya  Sure I looked for anywhere I changed code at all that was not simplify a type annotation. Since the error in ros2cli was seg fault I didn't have a ton of information so I had review all the changes I had made. Which only left one obvious change I had made. Here I had made function return None if goal_handle was None. I thought this was fine since the return type of the function included None. However I suspect  ros2cli might be doing some boundary checking which exposes that as an error. So I copied the exception it used to raise here instead. I'm hoping moving this exception will solve this issue.